src/graph/pickle.cc

```diff
@@ -107,8 +186,23 @@ DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroPickle")
 DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroUnpickle")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
     HeteroPickleStatesRef ref = args[0];
-    HeteroGraphPtr graph = HeteroUnpickle(*ref.sptr());
+    HeteroGraphPtr graph = ref->version ?
```

Comment:
Please use switch/if statement and check all the condition. (raise error by LOG(FATAL) if ref->version is not 0 or 1)